interface.go

```diff
@@ -19,79 +19,84 @@ type FieldValue interface{}
 // primary key, a partition key, a full row of a table, and so on.
 type FieldValuesByName map[FieldName]FieldValue
 
-// CacheKey is a slice a strings that joined together compose the key of a
-// cache entry.
-type CacheKey []string
+// CacheKey is the key of a cache entry.
+type CacheKey string
 
-// CacheValue is the value of a cache entry.
-type CacheValue string
+// CacheValue is the value of a cache entry. A nil slice means a cache miss, a
+// non nil slice means a cache hit even if it is empty.
+type CacheValue []byte
 
 // Table is the specification of a table.
 type Table struct {
 	// Name is the name of the table.
 	Name TableName `json:"name"`
-	// Fields specifies the types of all the fields of the table.
-	Fields FieldTypesByName `json:"fields"`
-	// PrimaryKey specifies the names of all the fields in the primary key
-	// of the table.
-	PrimaryKey []FieldName `json:"primaryKey"`
 	// PartitionKey specifies the names of all the fields in the partition
-	// key of the table.
+	// key of the table. The order in the slice matters.
 	PartitionKey []FieldName `json:"partitionKey"`
+	// PrimaryKey specifies the names of all the fields in the primary key
+	// of the table. It must contain the PartitionKey as prefix. The order
+	// in the slice matters.
+	PrimaryKey []FieldName `json:"primaryKey"`
+	// Values specifies the names of all the fields that are not in the
+	// primary key. The order in the slice matters.
+	Values []FieldName `json:"values"`
+	// Types specifies the types of all the fields of the table.
+	Types FieldTypesByName `json:"types"`
 	// PrimaryKeyCachePrefix is the prefix of all the keys in the primary
 	// key cache of the table. Each row in the table has the primary key as
 	// CacheKey and the remaining fields as CacheValue. The primary key is
 	// enabled if and only if the PrimaryKeyCachePrefix is not null. Users
 	// probably want to set a string that identifies a table and a version
 	// of the contents of the cache.
-	PrimaryKeyCachePrefix CacheKey `json:"primaryKeyCachePrefix"`
+	PrimaryKeyCachePrefix []CacheKey `json:"primaryKeyCachePrefix"`
 }
 
-// TableAccess is the interface of all types that represent an access to a
-// table.
-type TableAccess interface{}
+// RowAccess is the interface of all types that represent an access to a table.
+type RowAccess interface{}
 
-// TableCreate specifies the creation of a row in a table.
-type TableCreate struct {
+// RowCreate specifies the creation of a row in a table.
+type RowCreate struct {
 	// Table is the specification of the table.
 	Table Table
-	// FieldValues contains the primary key.
+	// FieldValues contains the primary key and the values.
 	FieldValues FieldValuesByName
 }
 
-// TableUpdate specifies the update of a row in a table.
-type TableUpdate struct {
+// RowRetrieve specifies the retrieval of one or several rows in a table.
+type RowRetrieve struct {
```

Comment:
Should we refer to Row(s) in plural?

Reply:
In English a plural noun used as an adjective is written in singular, right?